app/models/generic_object.rb

```diff
@@ -200,4 +201,10 @@ def _call_automate(method_name, *args)
     ws = MiqAeEngine.deliver(options)
     ws.root['method_result']
   end
+
+  def remove_go_from_all_related_services
+    ServiceResource.where(:resource => self).each do |resource|
+      remove_from_service(Service.find_by(:id => resource.service_id))
```

Comment:
You do not have to lookup the service as it is an association of the `resource` object: `remove_from_service(resource.service))`